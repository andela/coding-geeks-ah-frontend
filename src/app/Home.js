import React from 'react';

const Home = () => (
  <div>
    <p className="intro">Authors Haven by Coding Geeks</p>
  </div>
);

export default Home;
