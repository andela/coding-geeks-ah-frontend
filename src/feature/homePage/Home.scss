.App-header {
    background-color: #282c34;
    height: 6.2rem;
    font-size: 25px;
    color: white;
  }