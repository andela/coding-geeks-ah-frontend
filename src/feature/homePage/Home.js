import React from 'react';
import { withRouter } from 'react-router-dom';
import Nav from '../../app/routes/Nav';
import './Home.scss';

const Home = () => (
  <div>
    <header className="App-header">
      <Nav />
    </header>
    <p className="intro">Authors Haven by Coding Geeks</p>
  </div>
);

export default withRouter(Home);
