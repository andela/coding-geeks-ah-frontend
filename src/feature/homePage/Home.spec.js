import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from './Home';

const middlewares = [thunk, promiseMiddleware];
const mockStore = configureStore(middlewares);
const store = mockStore({
  isAuthenticated: true,
  social: jest.fn(),
  login: jest.fn(),
});
let wrapper;

describe('Render the Nav Bar', () => {
  it('Should render Home page successfully', async () => {
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
