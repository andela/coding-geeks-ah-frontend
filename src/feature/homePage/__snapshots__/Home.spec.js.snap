// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the Nav Bar Should render Home page successfully 1`] = `ReactWrapper {}`;
